app/bundles/DynamicContentBundle/Entity/DynamicContent.php

```diff
@@ -204,6 +222,11 @@ public static function loadValidatorMetaData(ClassMetadata $metadata): void
         $metadata->addPropertyConstraint('name', new NotBlank(['message' => 'mautic.core.name.required']));
         $metadata->addPropertyConstraint('content', new NoNesting());
 
+        $metadata->addPropertyConstraint('type', new NotBlank(['message' => 'mautic.core.name.required']));
+        $metadata->addPropertyConstraint('type', new TypeChoice());
```

Comment:
```suggestion
        $metadata->addPropertyConstraint('type', new Choice(['choices' => (new TypeList())->getChoices()]));
```
Remove both `\Mautic\DynamicContentBundle\Validator\Constraints\TypeChoice` and `\Mautic\DynamicContentBundle\Validator\Constraints\TypeChoiceValidator` as they don't make any sense. Moreover `\Mautic\DynamicContentBundle\Validator\Constraints\TypeChoice::ERROR_MESSAGE` is not translated.